### PR TITLE
Decoupling

### PR DESCRIPTION
**Pull request by lukasz-ligizynski**
So I started writing into monteway, but as the first step, I decoupling repository/use cases. I would know if it's okay and can I start going full monte. In the meantime, there was a small json gem change too :)

### CODE REVIEW

File: app/controllers/halls_controller.rb [hunk 1]
```diff
@@ -1,44 +1,47 @@
 # frozen_string_literal: true
 
 class HallsController < ApplicationController
-
   def index
-    halls = Hall.all
-    render json: HallSerializer.new(halls)
+    hall = repo.all
+    render json: HallSerializer.new(hall)
```
Comment (wmaciejak) [inline app/controllers/halls_controller.rb:6]: You can change to `render jsonapi: hall` and it will automatically find good serializer.

**Comment by wmaciejak** [comment]
I did not mention this approach with callbacks during our lessons, but it's also okay. Looks a bit different but if you understand it well, I'm really happy with this approach. It looks really cool 👍

**Comment by lukasz-ligizynski** [comment]
> I did not mention this approach with callbacks during our lessons, but it's also okay. Looks a bit different but if you understand it well, I'm really happy with this approach. It looks really cool 👍

Yeah, I know that there were no callbacks during lessons. But when I saw them in the blog I wanted to read more about them and know how they work :D So  I think I'm fine with them. Got some nice articles about them with examples of how to use them.

**Comment by wmaciejak** [comment]
To be honest - it's really good approach, you create really good communication flow directly in use cases 👍 

Anyway - ✔️

**Review by wmaciejak — APPROVED** [review]

**Pull request merged**